app/models/queue_item.rb

```diff
@@ -5,22 +5,45 @@ class QueueItem < ActiveRecord::Base
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
 
-  before_create :set_order
+  before_create :set_position
 
-  default_scope { order(:order) }
+  validates_numericality_of :position, only_integer: true, allow_blank: true
+  validates_uniqueness_of :video, scope: :user
+
+  def review
+    @review ||= Review.where(author: user, video: video).first
```

Comment:
Yea. In Rails 4, preferred way to search for a single record is `.find_by`. `where(...).first` was the Rails 3 way but currently is less efficient than former.